blueman/services/meta/SerialService.py

```diff
@@ -23,14 +32,31 @@ def serial_port_id(self, channel):
     def connected(self):
         return False
 
+    def on_file_changed(self, monitor, file, other_file, event_type, root_watcher_pid):
+        if event_type == Gio.FileMonitorEvent.DELETED:
+            dprint('%s got deleted' % file.get_path())
+        elif event_type == Gio.FileMonitorEvent.ATTRIBUTE_CHANGED and os.access(file.get_path(), os.R_OK | os.W_OK):
```

Comment:
Had a quick look and you could consider querying the attributes on the gfile instead of os module. I don't like the Cish api though so won't blame you to keep it as it is :smile: 

```
info = gfile.query_info('*', Gio.FileQueryInfoFlags.NONE) # * gets all attributes
can_write = info.get_attribute_boolean(Gio.FILE_ATTRIBUTE_ACCESS_CAN_WRITE)
```